apps/native/app/src/contexts/user-provider.tsx

```diff
@@ -0,0 +1,35 @@
+import { ApolloError } from '@apollo/client'
+import { PropsWithChildren, createContext, useContext } from 'react'
+import { GetProfileQuery, useGetProfileQuery } from '../graphql/types/schema'
+
+type UserContextType = {
+  user: GetProfileQuery['getUserProfile'] | null
+  refetch(): void
+  loading: boolean
+  error?: ApolloError
+}
+
+export const UserContext = createContext<UserContextType | undefined>(undefined)
+
+export const UserProvider = ({ children }: PropsWithChildren) => {
```

Comment:
Not sure if this provider is a good idea:
- It is not integrated into the auth store lifecycle afaics. Does it do a failing API request when I'm not logged in? When does it refetch? When I log in? When I switch users?
- It adds one more API request and load on our servers everytime users open the app, even when we don't need it.
- If we want this in a context, we should add it to the current userinfo request logic.

I think it's fine to fetch the user in document-reply. You can also pass the email from the reply mutation to onReplySuccess to show it on the document-communication screen.